conjure-java-core/src/integrationInput/java/com/palantir/product/EteBinaryServiceEndpoints.java

```diff
@@ -1,127 +1,229 @@
 package com.palantir.product;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.reflect.TypeToken;
 import com.palantir.conjure.java.undertow.lib.BinaryResponseBody;
 import com.palantir.conjure.java.undertow.lib.Endpoint;
-import com.palantir.conjure.java.undertow.lib.EndpointRegistry;
-import com.palantir.conjure.java.undertow.lib.Registrable;
-import com.palantir.conjure.java.undertow.lib.SerializerRegistry;
-import com.palantir.conjure.java.undertow.lib.Service;
-import com.palantir.conjure.java.undertow.lib.ServiceContext;
-import com.palantir.conjure.java.undertow.lib.internal.Auth;
-import com.palantir.conjure.java.undertow.lib.internal.BinarySerializers;
-import com.palantir.conjure.java.undertow.lib.internal.StringDeserializers;
+import com.palantir.conjure.java.undertow.lib.UndertowRuntime;
+import com.palantir.conjure.java.undertow.lib.UndertowService;
 import com.palantir.tokens.auth.AuthHeader;
 import io.undertow.server.HttpHandler;
 import io.undertow.server.HttpServerExchange;
+import io.undertow.util.HttpString;
+import io.undertow.util.Methods;
 import io.undertow.util.StatusCodes;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Deque;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import javax.annotation.Generated;
 
 @Generated("com.palantir.conjure.java.services.UndertowServiceHandlerGenerator")
-public final class EteBinaryServiceEndpoints implements Service {
+public final class EteBinaryServiceEndpoints implements UndertowService {
     private final UndertowEteBinaryService delegate;
 
     private EteBinaryServiceEndpoints(UndertowEteBinaryService delegate) {
         this.delegate = delegate;
     }
 
-    public static Service of(UndertowEteBinaryService delegate) {
+    public static UndertowService of(UndertowEteBinaryService delegate) {
         return new EteBinaryServiceEndpoints(delegate);
     }
 
     @Override
-    public Registrable create(ServiceContext context) {
-        return new EteBinaryServiceRegistrable(context, delegate);
+    public List<Endpoint> endpoints(UndertowRuntime runtime) {
+        return ImmutableList.of(
+                new PostBinaryEndpoint(runtime, delegate),
+                new GetOptionalBinaryPresentEndpoint(runtime, delegate),
+                new GetOptionalBinaryEmptyEndpoint(runtime, delegate),
+                new GetBinaryFailureEndpoint(runtime, delegate));
     }
 
-    private static final class EteBinaryServiceRegistrable implements Registrable {
+    private static final class PostBinaryEndpoint implements HttpHandler, Endpoint {
```

Comment:
now that these are static classes, could also consider making them package private and splitting into independent files. Will help deal with overall length of the code.

Reply:
Services are generated in user-defined packages, it's likely that user-defined code exists in the same package and can access package private components.